compositor_render/src/transformations/web_renderer/transformation.rs

```diff
@@ -0,0 +1,157 @@
+use std::{
+    collections::HashMap,
+    env,
+    error::Error,
+    io,
+    num::ParseIntError,
+    process::{self, Command},
+    rc::Rc,
+};
+
+use compositor_common::scene::{NodeId, Resolution, TransformationRegistryKey};
+use image::ImageError;
+
+use crate::renderer::{
+    texture::Texture,
+    transformation::{Transformation, TransformationParams},
+    WgpuCtx,
+};
+
+use super::http_client::{HttpClient, HttpClientError};
+
+pub struct WebRenderer {
+    http_client: HttpClient,
+    wgpu_ctx: Rc<WgpuCtx>,
+    renderer_process: process::Child,
+}
+
+impl Transformation for WebRenderer {
+    fn apply(
+        &self,
+        params: &TransformationParams,
+        _sources: &HashMap<NodeId, Texture>,
+        target: &Texture,
+    ) -> Result<(), Box<dyn Error>> {
+        let TransformationParams::String(url) = params else {
+            return Err(Box::new(WebRendererApplyError::InvalidParam));
+        };
+
+        let size = target.size();
+        let frame = self.http_client.get_frame(
+            url.to_owned(),
+            Resolution {
+                width: size.width as usize,
+                height: size.height as usize,
+            },
+        )?;
+
+        if !frame.is_empty() {
+            self.upload(&frame, target)?;
+        }
+
+        Ok(())
+    }
+
+    fn registry_key(&self) -> TransformationRegistryKey {
+        TransformationRegistryKey("web_renderer".to_owned())
+    }
+}
+
+impl Drop for WebRenderer {
+    fn drop(&mut self) {
+        self.renderer_process.kill().unwrap();
+    }
+}
+
+impl WebRenderer {
+    pub fn new(wgpu_ctx: Rc<WgpuCtx>) -> Result<Self, WebRendererNewError> {
+        let port: u16 = env::var("WEB_RENDERER_PORT")?.parse()?;
```

Comment:
why are we defining that port via env? 
- Easy approach would be to just hardcode the port.
- A bit more involved would be to pick free port and pass it to the electron when starting it.
- If we want to make it configurable, it should be passed inside our api, not via env.